adapter/slack.py

```diff
@@ -51,4 +53,27 @@ async def ask(self, request: Request):
                     raise HTTPException(status_code=400, detail=f"Slack API Error : {delete_error}")
 
             raise HTTPException(status_code=400, detail=f"Slack API Error : {e}")
+    
+    async def list_bots(self, request: Request):
+        data = await request.form()
+    
+        channel_id = data.get("channel_id")
+        
+        bot_responses = self.bot_controller.fetch_chatbots()
         
+        response_text = f"{len(bot_responses)} Active Bots:"
+        response_text += ''.join([f'\n- {bot_response.name} ({bot_response.id})' for bot_response in bot_responses])
+        
+        try:
+            response = self.app.client.chat_postMessage(channel=channel_id, text=response_text)
+            thread_ts = response["ts"]
+            return Response(status_code=200)
+
+        except SlackApiError as e:
+            if 'thread_ts' in locals():

```

Comment:
this line also not needed because there is no reply feature on list bot

Reply:
Same as discussion above, I've just removed the irrelevant codes.